node/web/src/lib.rs

```diff
@@ -119,18 +127,34 @@ async fn setup_node(
         .work_verifier_index(work_verifier_index.clone());
 
     // TODO(binier): refactor
-    if let Some(seed_nodes_url) = seed_nodes_url {
-        let peers = ::node::core::http::get_bytes(&seed_nodes_url)
-            .await
-            .expect("failed to fetch seed nodes");
-        node_builder.initial_peers(
-            String::from_utf8_lossy(&peers)
-                .split("\n")
-                .filter(|s| !s.trim().is_empty())
-                .map(|s| s.trim().parse().expect("failed to parse seed node addr")),
-        );
+    let mut all_raw_peers = seed_nodes_fixed.unwrap_or_default();
+    if let Some(seed_nodes_urls) = seed_nodes_urls {
+        for seed_nodes_url in seed_nodes_urls {
+            let peers = ::node::core::http::get_bytes(&seed_nodes_url).await;
+            match peers {
+                Ok(s) => {
+                    log::info!("Successfully fetched peers from {seed_nodes_url}");
+                    all_raw_peers.extend(String::from_utf8_lossy(&s).split("\n").map(String::from));
+                }
+                Err(e) => {
+                    log::error!("Failed to fetch peers from {seed_nodes_url}: {e}");
+                }
+            }
+        }
     }
 
+    node_builder.initial_peers(
+        all_raw_peers
+            .iter()
+            .map(|s| s.trim())
+            .filter(|s| !s.is_empty())
+            .flat_map(|s| s.parse().ok())
+            .map(|p| {
+                log::debug!("Using peer: {p:?}");
+                p
+            }),
```

Comment:
another nit
```suggestion
            .inspect(|p| log::debug!("Using peer: {p:?}")),
```
I'm assuming that `debug!` returns unit here...

Reply:
Great nit! I always forget about good old `inspect`!